caffeine/src/main/java/com/github/benmanes/caffeine/cache/AsyncCache.java

```diff
@@ -170,7 +181,7 @@ CompletableFuture<Map<K, V>> getAll(Iterable<? extends K> keys,
    * @param valueFuture the value to be associated with the specified key
    * @throws NullPointerException if the specified key or value is null
    */
-  void put(K key, CompletableFuture<? extends V> valueFuture);
+  void put(K key, CompletableFuture<? extends @Nullable V> valueFuture);
 
   /**
    * Returns a view of the entries stored in this cache as a thread-safe map. Modifications made to
```

Comment:
Sorry, I also meant to ask whether `@Nullable V` would make sense for the type argument inside `asMap()`. That, too, though, is unrelated to the main change.

Reply:
I think so because we always return the user's future and not any of our own dependent actions. There are a variety of technical reasons for this wrt correctness, but it means if their future resolves to null or an error state then it is observable. The cache will discard it by a dependent action.

Reply:
Thanks. I'm guessing this is another case in which `@Nullable` would be inconvenient, so I've changed this to also be null-unmarked with per-type annotations.